### ISSUE TITLE
Nightmare trying to install via recommended pypi-install method on Debian Stable (Buster)

### ISSUE THREAD

**Issue by TRSx80** [ultrabug/py3status#1916]
First off, I am actually really excited about using py3status, as it looks exactly like what I have been looking for. So thank you very much for sharing your work, and for continuing to maintain and improve it! It is much appreciated!

I am about to give up and just try to install via pip3, but since this method was recommended over doing that, I thought I would at least leave some feedback.

I will try and summarize my rough steps. After installing `python-stdeb` package in Debian, I first tried:
```
$ sudo pypi-install py3status
```
It seemed to mostly go alright, but I think something happened when byte-compiling `frame.py`:
```
byte-compiling /tmp/tmp9bsHMK/deb_dist/py3status-3.27/debian/python-py3status/usr/lib/python2.7/dist-packages/py3status/modules/frame.py to frame.pyc
  File "/usr/lib/python2.7/dist-packages/py3status/modules/frame.py", line 128
    composites[f'output_{o["name"]}'] = o["full_text"]
                                   ^
SyntaxError: invalid syntax
```
Also interesting to note, is that for some reason it seems to default to python2 (which is still the default on Debian Buster as far as I know).

So of course then it barfs a little later during setup stage:
```
Setting up python-py3status (3.27-1) ...
  File "/usr/lib/python2.7/dist-packages/py3status/modules/frame.py", line 128
    composites[f'output_{o["name"]}'] = o["full_text"]
                                   ^
SyntaxError: invalid syntax

dpkg: error processing package python-py3status (--install):
 installed python-py3status package post-installation script subprocess returned error exit status 101
Errors were encountered while processing:
 python-py3status
Traceback (most recent call last):
  File "/usr/bin/pypi-install", line 67, in <module>
    main()
  File "/usr/bin/pypi-install", line 59, in main
    subprocess.check_call(cmd, shell=True)
  File "/usr/lib/python2.7/subprocess.py", line 190, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command 'dpkg -i *.deb' returned non-zero exit status 1
```

Then here I may have gone off the rails trying to install with python3 by adding options for `py2dsc-deb` at end of command like:
```
sudo pypi-install py3status --with-python3 True --no-python2-scripts True
```
To which I got some errors about `Unmet build dependencies: python3-all`, so I installed that and kept going with different options but I am afraid I just got further and further off into the weeds.

I do have a (now sanitized) log of the whole gory process that I could post up somewhere if you feel it might be helpful?

But for now I figure I should probably stop flailing about in order to not do any more damage than I have already done. :/

**Comment by TRSx80** [comment]
I did some more testing this morning on a bone stock Debian XFCE VM, freshly upgraded to 10.3, just to make sure the problem wasn't in my environment.

```
# apt install python-stdeb
# pypi-install py3status
```
First I got:
```
Unmet build dependencies: dh-python
```
So then I did:
```
# apt install dh-python
```
...and re-ran:
```
# pypi-install py3status
```
But I still ended up with the same result as OP.

Then I realized there is a `python3-stdeb` package. So I did:
```
# apt purge python-stdeb
# apt autoremove
# apt install python3-stdeb
# pypi-install py3status
```
This resulted in:
```
bash: /usr/bin/pypi-install: No such file or directory
```
???

**Comment by TRSx80** [comment]
I ended up just doing:
```
pip3 install py3status
```
Which worked like a charm; no muss, no fuss. Recent versions of Debian (and Ubuntu) apparently now default to --user and install into `~/.local/bin/`. I wish I would have done that in the first place, it would have saved me about 24 hours of dorking around. :disappointed:

I am still willing to help diagnose the original issue as filed, or perhaps Docs should be updated as to recommended installation method on Debian, or...?

**Comment by lasers** [comment]
python-stdeb (0.8.5-1)- https://packages.debian.org/buster/python-stdeb. Released on 2015-02-18.

```python
composites[f'output_{o["name"]}'] = o["full_text"]  # SyntaxError: invalid syntax
```
... is a Python 3.6 feature. Released on December 23, 2016.

I dunno why you're getting... 
```bash
bash: /usr/bin/pypi-install: No such file or directory
```
Maybe you requires a new terminal after uninstalling/installing.

There is new version https://github.com/astraw/stdeb too, but yeah, use `pip3` instead.

**Comment by ultrabug** [comment]
Sorry for the reply delay @TRSx80 

First of all I want to extend a warm thank you for your positive attitude, kind words and taking the time to explain what you've been through. This is a rare gift you've giving us.

The documentation is the reflection of such journeys and keeping it actual is important.

I'm no debian user myself so I cannot argue on this. This method AFAIK was recommended by the kind debian developer who got the package in there so I'm pinging him in case he can take the time to shed a light on things

friendly ping @sdelafond

**Comment by sdelafond** [comment]
python3-stdeb is unfortunately a bit crippled because of
https://bugs.debian.org/890329. Here's what worked for me on buster:

```
apt install python3-stdeb python-stdeb
pypi-install py3status --with-python3 True --with-python2 False
```

**Comment by TRSx80** [comment]
> `apt install python3-stdeb python-stdeb`
> `pypi-install py3status --with-python3 True --with-python2 False`

Can confirm this worked for me in a VM. And by "worked" I mean it didn't crash out at any point. Seemed to be able to start py3status after it compiled or whatever. I didn't test thoroughly because I already got this working on my "real" desktop via pip3. But I did just want to report back at least that much.

Whether you want to keep this as a recommended installation method is up to you guys of course. But perhaps at least update the docs? Or maybe having this Issue is good enough for people to find, for now?

> First of all I want to extend a warm thank you for your positive attitude, kind words and taking the time to explain what you've been through. This is a rare gift you've giving us.

It's the very least I can do. And now I am enjoying py3status quite a lot since I got it up and running (via pip3)! :smile: Cheers, mate! :beers:
